qcodes/instrument_drivers/ZI/ZIUHFLI.py

```diff
@@ -970,16 +964,45 @@ def __init__(self, name: str, device_ID: str, **kwargs) -> None:
                                 val_mapping={'ON': 1, 'OFF': 0},
                                 vals=vals.Enum('ON', 'OFF') )
 
-            self.add_parameter('signal_output{}_enable'.format(sigout),
-                                label="Enable signal output's amplitude.",
-                                set_cmd=partial(self._sigout_setter,
-                                                sigout-1, 0,
-                                                outputampenable[sigout]),
-                                get_cmd=partial(self._sigout_getter,
-                                                sigout-1, 0,
-                                                outputampenable[sigout]),
-                                val_mapping={'ON': 1, 'OFF': 0},
-                                vals=vals.Enum('ON', 'OFF') )
+            if 'MF' in self.props['options']:
+                for modeout in range(1, 9):
+                    self.add_parameter('signal_output{}_amplitude{}'.format(sigout, modeout),
+                                       label='Signal output amplitude',
+                                       set_cmd=partial(self._sigout_setter,
+                                                       sigout - 1, 1, 'amplitudes', output_mode=modeout - 1),
+                                       get_cmd=partial(self._sigout_getter,
+                                                       sigout - 1, 1, 'amplitudes', output_mode=modeout - 1),
+                                       unit='V')
```

Comment:
I'd add a `docstring=...` to this parameter explaining that "the actual unit and representation is defined by `signal_output{}_ampdef` parameter". (same for the non_MF parameters)

Reply:
✔️